sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,189 @@
+        self._max_batch_size = eventhub_client.config.max_batch_size
+        self._receive_timeout = eventhub_client.config.receive_timeout
+        self._tasks: List[asyncio.Task] = []
+        self._id = str(uuid.uuid4())
+        self._partition_ids = None
+
+    @property
+    def id(self):
+        return self._id
+
+    async def start(self):
+        """Start the EventProcessor.
+            1. retrieve the partition ids from eventhubs
+            2. claim partition ownership of these partitions.
+            3. repeatedly call EvenHubConsumer.receive() to retrieve events and
+            call user defined PartitionProcessor.process_events()
+        """
+        log.info("EventProcessor %r is being started", self._id)
+        partition_ids = await self._eventhub_client.get_partition_ids()
+        self._partition_ids = partition_ids
+        claimed_list = await self._claim_partitions()
+        await self._start_claimed_partitions(claimed_list)
+
+    async def stop(self):
+        """Stop all the partition consumer
+
+        This method cancels tasks that are running EventHubConsumer.receive() for the partitions owned by this EventProcessor.
+        """
+        for task in self._tasks:
+            task.cancel()
+            # It's not agreed whether a partition manager has method close().
+        log.info("EventProcessor %r has been cancelled", self._id)
+        await asyncio.sleep(2)  # give some time to finish after cancelled
+
+    async def _claim_partitions(self):
+        partitions_ownership = await self._partition_manager.list_ownership(self._eventhub_name, self._consumer_group_name)
+        partitions_ownership_dict = dict()
+        for ownership in partitions_ownership:
+            partitions_ownership_dict[ownership["partition_id"]] = ownership
+
+        to_claim_list = []
+        for pid in self._partition_ids:
+            p_ownership = partitions_ownership_dict.get(pid)
+            if p_ownership:
+                to_claim_list.append(p_ownership)
+            else:
+                new_ownership = {"eventhub_name": self._eventhub_name, "consumer_group_name": self._consumer_group_name,
+                                 "owner_id": self._id, "partition_id": pid, "owner_level": OWNER_LEVEL}
+                to_claim_list.append(new_ownership)
+        claimed_list = await self._partition_manager.claim_ownership(to_claim_list)
+        return claimed_list
+
+    async def _start_claimed_partitions(self, claimed_partitions):
+        consumers = []
+        for partition in claimed_partitions:
+            partition_id = partition["partition_id"]
+            offset = partition.get("offset")
+            offset = offset or self._initial_event_position
+            consumer = self._eventhub_client.create_consumer(self._consumer_group_name, partition_id,
+                                                            EventPosition(str(offset)))
+            consumers.append(consumer)
```

Comment:
`consumers` does not seem to be used anywhere?

Reply:
trash code during development